app/src/main/java/org/piwigo/ui/accounts/ManageAccountsActivity.java

```diff
@@ -0,0 +1,225 @@
+     *   currently active account should be highlighted
+     *   */
+    ManageAccountsViewModel viewModel;
+    ActivityManageAccountsBinding binding;
+
+    private ListView userListView;
+
+    private UserAdapter mUserAdapter;
+
+
+    private class UserAdapter extends ArrayAdapter<User>{
+        public UserAdapter(List<User> objects) {
+            super(ManageAccountsActivity.this, 0, objects);
+        }
+        @Override
+        public View getView(int position, View view, ViewGroup parent) {
+            final User user = getItem(position);
+            if (view == null) {
+                LayoutInflater inflater = ManageAccountsActivity.this.getLayoutInflater();
+                view = inflater.inflate(R.layout.account_row, parent, false);
+            }
+            TextView username = (TextView) view.findViewById(R.id.username);
+            username.setText(user.username);
+            if(user.guest) {
+                username.setTypeface(Typeface.DEFAULT, Typeface.BOLD_ITALIC);
+            }else{
+                username.setTypeface(Typeface.DEFAULT, Typeface.BOLD);
+            }
+
+            TextView galleryUrlView = (TextView) view.findViewById(R.id.gallery_url);
+            galleryUrlView.setText(user.url);
+
+            ImageView imageView = (ImageView) view.findViewById(R.id.account_image);
+/* TODO fill image here (from favicon of gallery / User) */
+
+            return view;
+        }
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        AndroidInjection.inject(this);
+        setTheme(R.style.Theme_Piwigo);
+        super.onCreate(savedInstanceState);
+
+        viewModel = ViewModelProviders.of(this, viewModelFactory).get(ManageAccountsViewModel.class);
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_manage_accounts);
+        binding.setViewModel(viewModel);
+
+        /* TODO: isn't it bulls*** to set the title of the viewmodel from the activity and wouldn't this be better handled in the ViewModel directly? */
+        viewModel.title.set(getString(R.string.nav_manage_accounts));
+        viewModel.users.set(accountHelper.getUsers());
+        viewModel.setView(this);
+
+        Toolbar toolbar = (Toolbar) findViewById(R.id.account_toolbar);
+        setSupportActionBar(toolbar);
+
+        userListView = (ListView) findViewById(R.id.user_list);
+        mUserAdapter = new UserAdapter(viewModel.users.get());
+        userListView.setAdapter(mUserAdapter);
+
+        viewModel.users.addOnPropertyChangedCallback(new Observable.OnPropertyChangedCallback() {
+            @Override
+            public void onPropertyChanged(Observable sender, int propertyId) {
+                ManageAccountsActivity.this.mUserAdapter.notifyDataSetChanged();
+            }
+        });
+        viewModel.users.set(accountHelper.getUsers());
+
+        userListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
```

Comment:
Might be a better way to handle events with data binding.

Reply:
same question: Do you have an example?

Reply:
after going down to the lambda-question I think I got it.